src/Mvc/Mvc.Core/src/ProducesResponseTypeAttribute.cs

```diff
@@ -57,10 +87,37 @@ public ProducesResponseTypeAttribute(Type type, int statusCode)
         /// <value></value>
         internal bool IsResponseTypeSetByDefault { get; }
 
+        // Internal for testing
+        internal MediaTypeCollection ContentTypes => _contentTypes;
+
         /// <inheritdoc />
         void IApiResponseMetadataProvider.SetContentTypes(MediaTypeCollection contentTypes)
         {
-            // Users are supposed to use the 'Produces' attribute to set the content types that an action can support.
+            contentTypes.Clear();
```

Comment:
Let me do a quick sanity test on this. I don't super remember how this gets used, but I'd like to make sure we have additional MVC integration tests for this now that we're enabling this.

Reply:
Sure. From what I could tell, this gets used like an `out var` but it is tricky to reason about.

https://github.com/dotnet/aspnetcore/blob/709038b45159227a422807f26d6d41dd7707ea10/src/Mvc/Mvc.ApiExplorer/src/DefaultApiDescriptionProvider.cs#L439-L442